### DISCUSSION TITLE
Mobile app

### DISCUSSION THREAD

**Discussion by Ymirke** [twentyhq/twenty#6273] [Ideas]
Hey, 

I'm sure this is something you've talked about internally.

Would be valuable to be able to add people and opportunities on the go. 

Could make sense to wait for 1.x when things are more stable.

**Comment by JarWarren** [comment]
The team has spoken about this before.

In Discord they mentioned possibly working on it next year. The app will be published under their name but they plan to have a mechanism for pointing it at a self-hosted instance. Obviously, plans can change, be postponed or scrapped tho.

**Comment by ReessKennedy** [comment]
Testing Twenty and enjoying but also use Copper which also has a great Gmail integration Ans a mobile app which really is important for a mobile sales force.

**Comment by nonsonosperti** [comment]
Are there any updates on this. Has it been planned?

**Comment by Lejouz** [comment]
Very important feature. Will extend the user base dramatically. Without mobile app many companies can't use this great CRM.

**Reply by dataCenter430** [reply]
Hi, @Lejouz  are there any progress on it?
I'd love to have the chance to contribute to it.

**Comment by adlucio87** [comment]
Hi everyone! 👋

I've been following this thread and decided to 
build it myself. TwentyMobile is a native 
iOS/Android app for Twenty CRM — open source 
(AGPL-3.0), self-hosted friendly, zero data 
collection.

Current features:
✓ Contacts, Companies, Tasks, Notes
✓ Today Screen — daily operations hub
✓ Business card scanner (on-device OCR)
✓ Voice-to-text notes
✓ Push notifications for task reminders
✓ Dark/Light theme

Looking for beta testers!
IOS TestFlight: https://testflight.apple.com/join/PWG5Tekq
Android: https://play.google.com/store/apps/details?id=com.luciosoft.pocketcrm
Site: https://twentymobilecrm.luciosoft.it/
GitHub: https://github.com/adlucio87/TwentyMobile

@dataCenter430 — contributions welcome!

**Reply by fendle** [reply]
Hi,

that's a great news and installed it directly and got the following error when I try to enter the API key

![image](https://github.com/user-attachments/assets/dec18390-579f-4c9e-bd6d-e3de11a0a3e7)

**Reply by adlucio87** [reply]
Hey! Thanks for testing it out so quickly!

That error is actually a direct response from your Twenty backend's GraphQL server rejecting the initial login query. Unfortunately, the app's UI truncated the string, hiding the actual cause (e.g., schema changes or missing token scopes).

I've just pushed a new build (v1.0.1) that disables the text truncation and adds Sentry capture for these network exceptions.

Could you give it another spin with the updated version and paste the full error output here? That will help us pinpoint exactly what the Twenty Server is complaining about and apply a fix right away!

**Reply by fendle** [reply]
Hi, it's right now not available in Testflight. I assume it needs to be approved

**Reply by adlucio87** [reply]
Yes is in approval process

**Reply by fendle** [reply]
Hi @adlucio87, I still have the issue

**Reply by adlucio87** [reply]
@fendle thanks, now I have the errore and understand the cause. I will fix soon.

**Reply by adlucio87** [reply]
@fendle I released an update that should fix the problem.

**Reply by fendle** [reply]
I still get the error. It's 1.0.1 build 16?

**Reply by adlucio87** [reply]
Could you please test the latest, I released it just now, version 18. I can't understand why u get this error from graphql service me. In my server self hosted does not appear. Witch twenty CRM version are you using? Thanks

**Reply by fendle** [reply]
Hi, now I got one page further but get a OperationException with "This Query cannot be executed....". Maybe you contact me via Discord? Search for "fendle.".

**Reply by adlucio87** [reply]
hi fendle, I tried searching for you on Discrod but I couldn't get in touch. If you want, you can also email me. However, with the latest update released, the problem has been resolved; it was related to a limitation of the Twenty-CRM servers on multi-content queries. Let me know if now works!

**Reply by fendle** [reply]
Hi @adlucio87, I will try to contact you by e-mail. But I already see, the biggest point will be that's not personslized for an user. For example, I would like to see only the tasks which are assigned to me.

**Reply by adlucio87** [reply]
Hi @fendle,
it’s better to contact me via email. The app is currently in its main release. The goal is to keep it simple and user-friendly, with more features to be added soon.

**Comment by Ymirke** [comment]
Wow, kind of funny to google the same thing a couple of years later to see that I opened this thread. 

With progress in AI and how far twenty has come I hope this picks up more steam.

**Reply by Ymirke** [reply]
Also, with the amount of engagement for open-source on this - could it make sense for the twenty team to lay out some guidelines / base components and then have the open source community mostly build it?

This might make sense since building fast with AI is trivial as long as you have API endpoints and how to render it. 

PR reviews with some agent setups could make good feedback fast to contributors too. 

Just some thoughts.

**Reply by will-helio** [reply]
I honestly would be more than happy to contribute to this... We use TwentyCRM for some of our clients and we/they all love it, but the no mobile app is restrictive af... It NEEDS to happen yesterday.

**Comment by mkmike000** [comment]
PLEASE 🙏 TWENTY20!!! MAKE THIS HAPPEN!!! Just lay the foundations and let the power of the community build it out.
